helper/boltdd/boltdd.go

```diff
@@ -353,6 +353,35 @@ func (b *Bucket) Get(key []byte, obj interface{}) error {
 	return nil
 }
 
+// Iterate iterates each key in Bucket b that starts with prefix. fn is called on
+// the key and msg-pack decoded value. If prefix is empty or nil, all keys in the
+// bucket are iterated.
+func Iterate[T any](b *Bucket, prefix []byte, fn func([]byte, T)) error {
```

Comment:
It's a little strange to not let `fn` return a `bool` or `error` indicating whether or not iteration can continue. Shouldn't be a hard change if we ever need it though.